x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtSignatureValidator.java

```diff
@@ -370,38 +383,53 @@ default void validateSignature(final SignedJWT jwt, final List<JWK> jwks) throws
         // - If JWT alg is HS256, all are valid for a JWT HS256 signature. Don't ignore any HMAC JWKs.
         // - If JWT alg is HS384, only 384, 400, 512, and 1000 are valid for a JWT HS384 signature. Ignore two HMAC JWKs.
         // - If JWT alg is HS512, only 512 and 1000 are valid for a JWT HS512 signature. Ignore four HMAC JWKs.
-        final List<JWK> jwksStrength = jwksAlg.stream().filter(j -> JwkValidateUtil.isMatch(j, alg.getName())).toList();
-        logger.debug("JWKs [{}] after Algorithm [{}] match filter.", jwksStrength.size(), alg);
+        final List<JWK> jwksConfigured = jwksAlg.stream().filter(j -> JwkValidateUtil.isMatch(j, alg.getName(), tracer)).toList();
+        tracer.append("[{}] JWKs remain after filtering for configured algorithms.", jwksConfigured.size());
+        tracer.flush();
 
         // No JWKs passed the kid, alg, and strength checks, so nothing left to use in verifying the JWT signature
-        if (jwksStrength.isEmpty()) {
-            throw new ElasticsearchException("Verify failed because all " + jwks.size() + " provided JWKs were filtered.");
+        if (jwksConfigured.isEmpty()) {
+            throw new ElasticsearchException(
+                "Signature verification was not attempted since there are not any JWKs "
+                    + "available after filtering for incompatible keys."
+            );
         }
 
-        for (final JWK jwk : jwksStrength) {
+        int attempt = 0;
+        int maxAttempts = jwksConfigured.size();
+        tracer.append("Attempting to verify signature for JWT [{}] against [{}] possible JWKs.", getSafePrintableJWT(jwt), maxAttempts);
+        for (final JWK jwk : jwksConfigured) {
+            attempt++;
             if (jwt.verify(createJwsVerifier(jwk))) {
-                logger.trace(
-                    "JWT signature validation succeeded with JWK kty=[{}], jwtAlg=[{}], jwtKid=[{}], use=[{}], ops=[{}]",
-                    jwk.getKeyType(),
-                    jwk.getAlgorithm(),
+                tracer.append(
+                    "Attempt [{}/{}] -> JWT signature verification succeeded with jwk/kid=[{}], jwk/alg=[{}], jwk/kty=[{}], jwk/use=[{}], "
+                        + "jwk/key_ops=[{}]",
+                    attempt,
+                    maxAttempts,
                     jwk.getKeyID(),
+                    jwk.getAlgorithm(),
+                    jwk.getKeyType(),
                     jwk.getKeyUse(),
                     jwk.getKeyOperations()
```

Comment:
I am not sure whether there are benefits to log all these information about JWK besides its keyId?